core/bug_api.php

```diff
@@ -1402,6 +1402,8 @@ function bug_delete( $p_bug_id ) {
 	$t_query = 'DELETE FROM {bug} WHERE id=' . db_param();
 	db_query( $t_query, array( $c_bug_id ) );
 
+	event_signal( 'EVENT_BUG_DELETED', array( $c_bug_id ) );
```

Comment:
According to current behaviour, and documentation, this event is triggered before deleting the bug and related items.

In this function, we already have 'issue_delete_validate' and 'issue_delete_notify' custom functions.
I wonder if we should have two events: one for validation, and one for notification, like we have for bug create and update. 
If a plugin must perform some action at bug deletion, it should be done after all other plugins have validated the action.
This scenario is less common, in my opinion,  that the update events one, but still worth considering.

On the other hand, any one of those events, should be triggered before the actual delete of data. Otherwise the information would not be available for plugins to do do their stuff.

For a simple approach here, i'd just place the current event in the same place as 'issue_delete_validate'